### PR TITLE
[ContextualSaveBar] hookify

### PR DESCRIPTION
**Pull request by sijad**
### WHY are these changes introduced?

related to #1995

### WHAT is this pull request doing?

change `ContextualSaveBar` to use react hooks

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

**Comment by sijad** [comment]
I'm not sure why `percy/polaris-react` failed. it seems there's a different with "Small-sized drop zone" but I didn't changed anything related to drobzone or any styles at all.

File: src/components/Frame/components/ContextualSaveBar/tests/ContextualSaveBar.test.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
-import {mountWithAppProvider, trigger} from 'test-utilities/legacy';
+import {act, mountWithAppProvider, trigger} from 'test-utilities/legacy';
 import {Button, Image} from 'components';
-import ContextualSaveBar from '../ContextualSaveBar';
+import {ContextualSaveBar} from '..';
```
Comment (AndrewMusgrave) [inline src/components/Frame/components/ContextualSaveBar/tests/ContextualSaveBar.test.tsx]:
```suggestion
import {ContextualSaveBar} from '../ContextualSaveBar';
```

Comment (sijad) [reply src/components/Frame/components/ContextualSaveBar/tests/ContextualSaveBar.test.tsx]: there are few other places that this occurred:

https://github.com/Shopify/polaris-react/blob/cda89db560c4a5d67045db888abf60978faf2c11/src/components/Backdrop/tests/Backdrop.test.tsx#L3
https://github.com/Shopify/polaris-react/blob/cda89db560c4a5d67045db888abf60978faf2c11/src/components/Banner/tests/Banner.test.tsx#L14
https://github.com/Shopify/polaris-react/blob/cda89db560c4a5d67045db888abf60978faf2c11/src/components/PageActions/tests/PageActions.test.tsx#L3
...

do you want me to send a PR for those too?

Comment (dleroux) [reply src/components/Frame/components/ContextualSaveBar/tests/ContextualSaveBar.test.tsx]: That would be great. Thanks for doing this!

**Comment by BPScott** [comment]
Thank you very much for contributing @sijad!

**Pull request merged**